### PR TITLE
Save regexp Pattern allocation on config name renaming

### PR DESCRIPTION
**Pull request by franz1981**
This is addressing a small regression in case some old names are used, saving allocating (for each replacement) a whole new regexp `Pattern` and creating a reusable function which does it "right" if compared to `String:.replaceFirst`.

I've than removed the method reference lambda generation to match what we do elsewhere, but I have not a strong opinion here, considered that it mostly affect RSS footprint while the application doesn't do anything interesting (sadly a case still valuable in public benchmarks :/ ) - see https://github.com/quarkusio/quarkus/issues/36204#issuecomment-1752989061  for more info at `java/lang/invoke/LambdaForm.compileToBytecode`.
In JDK 21 it's likely lambdas RSS footprint is reduced and who knows for later releases, so, as said, I've performed the change here, but I could have made the interceptors to implements `Function<String, String` and just pass `this` to their `super` constructor too, not getting lambdas nor anonymous class loaded, at all.

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkus/runtime/util/StringUtil.java [hunk 1]
```diff
@@ -13,6 +13,20 @@ public final class StringUtil {
     private StringUtil() {
     }
 
+    public static String changePrefix(String name, String oldPrefix, String newPrefix) {
+        if (!name.startsWith(oldPrefix)) {
+            return name;
+        }
+        // don't use here String::replaceFirst because it uses regex
```
Comment (gsmet) [inline core/runtime/src/main/java/io/quarkus/runtime/util/StringUtil.java:20]: Ah f**k. I thought `replaceFirst` was like `replace` and not using regexp :/

**Review by gsmet — CHANGES_REQUESTED** [review]
I agree with your change but the order of arguments is not always the same and I think you copy/pasted things a bit too much.

I will adjust and force push.

**Review by gsmet — APPROVED** [review]
I amended your commit to make sure the method was always called with the parameters in the correct order (relocate and fallback are symmetrical).

I also added a small test for your util as better to have it tested.

File: core/runtime/src/main/java/io/quarkus/runtime/util/StringUtil.java (hunk 1 above)
Comment (franz1981) [reply core/runtime/src/main/java/io/quarkus/runtime/util/StringUtil.java:20]: Yeah, I was surprised as well! Said that, you know that this change I'm pushing is a drop in the ocean which value is mostly on benchs as far as I can tell :/ 
And likely, the renaming will goes away in few releases too

File: core/runtime/src/test/java/io/quarkus/runtime/util/StringUtilTestCase.java [hunk 2]
```diff
@@ -98,4 +98,12 @@ public void testLowerCaseNoSuffix() {
         result = join(withoutSuffix(lowerCase(camelHumpsIterator("SomeRootClass")), "config", "class"));
         assertEquals("someroot", result);
     }
+
+    @Test
+    public void testChangePrefix() {
+        assertEquals("quarkus.new-prefix.configuration-property", StringUtil
```
Comment (franz1981) [inline core/runtime/src/test/java/io/quarkus/runtime/util/StringUtilTestCase.java:104]: nice one!

File: extensions/oidc-client-reactive-filter/runtime/src/main/java/io/quarkus/oidc/client/reactive/filter/runtime/OidcClientReactiveFilterConfigFallbackInterceptor.java [hunk 3]
```diff
@@ -10,16 +13,14 @@ public class OidcClientReactiveFilterConfigFallbackInterceptor extends FallbackC
...
-        super(OidcClientReactiveFilterConfigFallbackInterceptor::rename);
-    }
-
-    private static String rename(String originalName) {
-        if (!originalName.startsWith(NEW_PREFIX)) {
-            return originalName;
+    private static final Function<String, String> RENAME_FUNCTION = new Function<String, String>() {
+        @Override
+        public String apply(String s) {
+            return StringUtil.changePrefix(s, NEW_PREFIX, OLD_PREFIX);
```
Comment (franz1981) [inline extensions/oidc-client-reactive-filter/runtime/src/main/java/io/quarkus/oidc/client/reactive/filter/runtime/OidcClientReactiveFilterConfigFallbackInterceptor.java:19]: @gsmet My ide has been smarter than me and I fix it back (making it wrong) later on, uff :/ many thanks to have found it!

**Comment by franz1981** [comment]
Thanks @gsmet for the review :pray:

**Comment by gsmet** [comment]
Thanks, good catch!

**Pull request merged**